std/test.glu

```diff
@@ -38,6 +40,22 @@ let assert_neq l r : [Show a] -> [Eq a] -> a -> a -> Eff [| writer : Test | r |]
     if l /= r then wrap ()
     else tell (Cons ("Assertion failed: " <> show l <> " == " <> show r) Nil)
 
+let assert_ok res : [Show e] -> Result e a -> Eff [| writer : Test | r |] () =
+    match res with
+    | Ok _ -> wrap ()
```

Comment:
`Could return the value instead of `()`

Reply:
I considered that, but then what would the failing branch return?

Reply:
Ah, right. Because the error is just written, there is no error thrown. I think it would make sense to move these asserts into `std.effect.error` and have them throw instead. Then make tests instead catch errors and write them. The current setup is kind of strange in that it continues even if an assert fails.

That's another PR though as there might be fallout from such a change.